packages/server/tests/acceptance/htsPrecompile.spec.ts

```diff
@@ -34,60 +34,161 @@ describe('HTS Precompile Acceptance Tests', async function() {
+    accounts[0] = await servicesNode.createAliasAccount(60, relay.provider);
     accounts[1] = await servicesNode.createAliasAccount(30, relay.provider);
     accounts[2] = await servicesNode.createAliasAccount(30, relay.provider);
 
-    baseHTSContract = await deployBaseHTSContract();
-    HTSTokenContract = await createHTSToken();
+    BaseHTSContractAddress = await deployBaseHTSContract();
+    HTSTokenContractAddress = await createHTSToken();
+    NftHTSTokenContractAddress = await createNftHTSToken();
   });
 
   async function deployBaseHTSContract() {
     const baseHTSFactory = new ethers.ContractFactory(BaseHTSJson.abi, BaseHTSJson.bytecode, accounts[0].wallet);
-    const baseHTS = await baseHTSFactory.deploy();
+    const baseHTS = await baseHTSFactory.deploy({gasLimit: 10000000});
     const { contractAddress } = await baseHTS.deployTransaction.wait();
 
-    return new ethers.Contract(contractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    return contractAddress;
   }
 
   async function createHTSToken() {
-    const tx = await baseHTSContract.createToken(accounts[0].wallet.address, {
-      value: ethers.BigNumber.from('20000000000000000000'),
-      gasLimit: 1000000
+    const baseHTSContract = new ethers.Contract(BaseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = await baseHTSContract.createFungibleTokenPublic(accounts[0].wallet.address, {
+      value: ethers.BigNumber.from('10000000000000000000'),
+      gasLimit: 10000000
     });
     const { tokenAddress } = (await tx.wait()).events.filter(e => e.event = 'CreatedToken')[0].args;
 
-    return new ethers.Contract(tokenAddress, ERC20MockJson.abi, accounts[0].wallet);
+    return tokenAddress;
+  }
+
+  async function createNftHTSToken() {
+    const baseHTSContract = new ethers.Contract(BaseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = await baseHTSContract.createNonFungibleTokenPublic(accounts[0].wallet.address, {
+      value: ethers.BigNumber.from('10000000000000000000'),
+      gasLimit: 10000000
+    });
+    const { tokenAddress } = (await tx.wait()).events.filter(e => e.event = 'CreatedToken')[0].args;
+
+    return tokenAddress;
   }
 
   it('should associate to a token', async function() {
-    const baseHTSContractReceiverWalletFirst = new ethers.Contract(baseHTSContract.address, BaseHTSJson.abi, accounts[1].wallet);
-    expect(baseHTSContractReceiverWalletFirst.associateTokenTo(accounts[1].wallet.address, HTSTokenContract.address, { gasLimit: 10000000 }))
-      .to.not.be.reverted;
+    const baseHTSContractOwner = new ethers.Contract(BaseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const txCO = await baseHTSContractOwner.associateTokenPublic(BaseHTSContractAddress, HTSTokenContractAddress, { gasLimit: 10000000 });
+    expect((await txCO.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
```

Comment:
nit: status code `22` can be replaced with the const var `TX_SUCCESS_CODE`

Reply:
About to merge this to unblock you so you can make this modification in your PR on top of this